src/query/block/container.go

```diff
@@ -0,0 +1,642 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package block
+
+import (
+	"errors"
+	"time"
+
+	"github.com/m3db/m3/src/query/ts"
+	xerrors "github.com/m3db/m3/src/x/errors"
+)
+
+var (
+	errMismatchedStepIter   = errors.New("container step iter has mismatched step size")
+	errMismatchedUcStepIter = errors.New("unconsolidated container step iter has mismatched step size")
+)
+
+type containerBlock struct {
+	err    error
+	blocks []Block
+}
+
+func newContainerBlock(blocks []Block) AccumulatorBlock {
+	return &containerBlock{
+		blocks: blocks,
+	}
+}
+
+// NewContainerBlock creates a Container block.
+func NewContainerBlock(initialSize int) AccumulatorBlock {
+	initial := make([]Block, 0, initialSize)
+	return newContainerBlock(initial)
+}
+
+func (b *containerBlock) AddBlock(bl Block) error {
+	if b.err != nil {
+		return b.err
+	}
+
+	b.blocks = append(b.blocks, bl)
+	return nil
+}
+
+func (b *containerBlock) Close() error {
+	multiErr := xerrors.NewMultiError()
+	multiErr.Add(b.err)
+	for _, bl := range b.blocks {
+		multiErr.Add(bl.Close())
```

Comment:
This is because of our silly API, but this should be:
multiErr = multiErr.Add(bl.Close())

Reply:
Oops, fixed